src/error.cs

```diff
@@ -11,51 +11,63 @@
 
 using System;
 using System.Collections.Generic;
+using System.Reflection;
 
 using ProductException=BindingException;
 
 // Error allocation: the errors are listed (and documented) in $(TOP)/docs/website/generator-errors.md
 
 public class BindingException : Exception {
 	
-	public BindingException (int code, bool error, string message) :
-		base (message)
+	public BindingException (int code, bool error) :
+		base (GetMessage(code))
 	{
 		Code = code;
 		Error = error || ErrorHelper.GetWarningLevel (code) == ErrorHelper.WarningLevel.Error;
 	}
 
-	public BindingException (int code, string message, params object[] args) : 
-		this (code, false, message, args)
+	public BindingException (int code, params object[] args) : 
+		this (code, false, args)
 	{
 	}
 
-	public BindingException (int code, bool error, string message, params object[] args) : 
-		this (code, error, null, message, args)
+	public BindingException (int code, bool error, params object[] args) : 
+		this (code, error, null, args)
 	{
 	}
 
-	public BindingException (int code, bool error, Exception innerException, string message, params object[] args) : 
-		base (String.Format (message, args), innerException)
+	public BindingException (int code, bool error, Exception innerException, params object[] args) : 
+		base (String.Format (GetMessage(code), args), innerException)
 	{
 		Code = code;
 		Error = error || ErrorHelper.GetWarningLevel (code) == ErrorHelper.WarningLevel.Error;
 	}
 
+	static string GetMessage (int code)
+	{
+		Type resourceType = Type.GetType ("bgen.Resources");
+		string errorCode = string.Format ("BI{0:0000}", code);
+		PropertyInfo prop = resourceType.GetProperty (errorCode, BindingFlags.NonPublic |
+		    BindingFlags.Static |
+		    BindingFlags.GetProperty);
+		var errorMessage = (String) prop.GetValue (null);
+		return errorMessage;
+	}
```

Comment:
That needs some null checks - e.g. if there's a typo and the message does not exists then the customer will get an error that's even harder to debug (for them and us).

e.g. if `prop` is null you'll end with ` BI0000` because of an NRE
and not with the original error code (without a description)

Maybe we should have a default message to ask for an issue to be filled when this happens, e.g.

```BIxxxx: Sorry the error message for this code (xxxx) could not be found. Please report this missing message using GitHub...`